sqldelight-gradle-plugin/src/main/kotlin/app/cash/sqldelight/gradle/SqlDelightDatabase.kt

```diff
@@ -10,22 +10,37 @@ import app.cash.sqldelight.gradle.squash.MigrationSquashTask
 import groovy.lang.GroovyObject
 import org.gradle.api.GradleException
 import org.gradle.api.Project
+import org.gradle.api.file.DirectoryProperty
 import org.gradle.api.internal.catalog.DelegatingProjectDependency
+import org.gradle.api.provider.ListProperty
+import org.gradle.api.provider.Property
 import org.gradle.api.provider.Provider
 import java.io.File
+import javax.inject.Inject
 
-class SqlDelightDatabase(
+abstract class SqlDelightDatabase @Inject constructor(
   val project: Project,
   var name: String,
```

Comment:
One last thing, why is name writeable?
```suggestion
  val name: String,
```

Reply:
Updated. I'd left it as is, but you're right, doesn't seem like it needs to be writeable anymore.

Reply:
Yeah, changing the name of an object inside a NamedCollection isn't a good idea I guess.